### DISCUSSION TITLE
what changes do I need to do to print acro fields name only>

### DISCUSSION THREAD

**Discussion by kharabela** [pdfcpu/pdfcpu#720] [Q&A]
Below is my code to print form fields in acro form

// Displays a list of form fields in the source pdf.
func DisplayFormFields() {
	InFile, err := os.Open("pdffill/sample.pdf")
	if err != nil {
		fmt.Println(err)
	}
	ss, err := pdfcpu.FormFields(InFile, pdfcpu.LoadConfiguration())

	for _, s := range ss {

		fmt.Println(s)
	}
	InFile.Close()
}

My output is as
{[1] false 0 67 Name   }
{[1] false 3 68 Dropdown2 Jan Jan Jan,Feb,Mar,Apr,May,Jun,Jul,Aug,Sep,Oct,Nov,Dec}
{[1] false 3 71 Dropdown1 1 1 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31}
{[1] false 3 72 Dropdown3 2012 2012 2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025}
{[1] false 0 73 Address   }
{[1] false 2 74 Check Box1   }
{[1] false 2 76 Check Box2   }
{[1] false 2 77 Check Box3   }
{[1] false 2 78 Check Box4   }
{[1] false 0 79 Text5   }
{[1] false 5 7 Group6   0,1,2,3}
{[1] false 0 84 Text6   }
{[1] false 0 6 Button7   }

I wanto print only merge fields as in acrofield name only such as Address,Name

**Comment by hhrutter** [comment]
Please refer to **cli.ListFormFieldsFile** to see how the `pdfcpu form list`  command works.

**Reply by kharabela** [reply]
I want to use API only.what should I refer?

**Comment by hhrutter** [comment]
The api package returns a struct containing the field information and the cli package uses that for formatted output.
I pointed you to cli.ListFormFieldsFile because there you have the mapping from the struct to a list.
**ID** or **Name** is probably what you are looking for.

For further explanation please refer to the PDF specification.
Thank you!
